File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        set<int> numsSet(nums.begin(), nums.end());
        int longest = 0;
        for (int num: numsSet) {
            if (numsSet.find(num-1) == numsSet.end()) {
                int length = 1;
                while (numsSet.find(num+length) != numsSet.end())
                {
                    length++;
                }
                longest = max(length, longest);
            }
        }
        return longest;
    }
};

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-0.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        n = len(nums)
        dp = [1 for i in range(n)]
        res = 1
        for i in range(n-1, -1, -1):
            mx = 0
            for j in range(i+1, n):
                if nums[i] < nums[j]:
                    mx = max(dp[j], mx)
            dp[i] = dp[i] + mx
            res = max(dp[i], res)
        return res
        
